File: DailyLeetcode/14Aug/Largest3SameDigit.py
# Time-Complexity-O(n)
# Space-Complexity-O(n)
from collections import Counter
from typing import List
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        triplet=[]
        for i in range(len(num)-2):
            if num[i]==num[i+1]==num[i+2]:
                triplet.append(num[i:i+3])
        if not triplet:
            return ""
        counter=Counter(triplet)
        return max(counter.keys())
                

File: DailyLeetcode/14Aug/Largest3SameDigit.cpp
// Time Complexity:O(n)
// Space Complexity:O(1)

class Solution {
public:
    string largestGoodInteger(string num) {
        int n = num.length();
        string res = "";
        for (int i = 0; i + 2 < n; i++) {
            if (num[i] == num[i + 1] && num[i + 1] == num[i + 2] &&
                num[i] == num[i + 2]) {
                string cur = num.substr(i, 3);
                if (cur > res) {
                    res = cur;
                }
            }
        }
        return res;
    }
};